src/trajectory_generator.cpp

```diff
@@ -47,11 +47,12 @@ TrajectoryGenerator::TrajectoryGenerator(uint num_dof, double timestep, double d
 {
   // Upsample if num. waypoints would be short. Helps with accuracy
   upsample();
+  auto max_num_waypoints = desired_duration / upsampled_timestep_;
```

Comment:
Prefer not to use auto here. It's not clear if it will end up being a double or a size_t or something